tests/functional/test_hooks/test_uvicorn.py

```diff
@@ -0,0 +1,60 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2005-2020, PyInstaller Development Team.
+#
+# Distributed under the terms of the GNU General Public License (version 2
+# or later) with exception for distributing the bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#
+# SPDX-License-Identifier: (GPL-2.0-or-later WITH Bootloader-exception)
+#-----------------------------------------------------------------------------
+
+from PyInstaller.utils.tests import importorskip
+
+
+@importorskip('uvicorn')
+def test_uvicorn(pyi_builder):
+    """
+    Check if uvicorn builds correctly.
+    """
+    pyi_builder.test_source(
+        """
+        import uvicorn
+        import sys
+
+        # as shown on:
+        # https://www.uvicorn.org/#requests-responses
+
+        async def app(scope, receive, send):
+            '''
+            Echo the method and path back in an HTTP response.
+            '''
+            assert scope['type'] == 'http'
+
+            body = f'Received {scope["method"]} request to {scope["path"]}'
+            await send({
+                'type': 'http.response.start',
+                'status': 200,
+                'headers': [
+                    [b'content-type', b'text/plain'],
+                ]
+            })
+            await send({
+                'type': 'http.response.body',
+                'body': body.encode('utf-8'),
+            })
+
+        if __name__ == '__main__':
+            # TODO: is sys.exit needed?
+            #sys.exit(uvicorn.run(
```

Comment:
Is `sys.exit` needed? If not, please remove it from the code.

Reply:
Also, just noticed. Please change to `.format()`, not f-strings. They are only supported in python 3.6+ and we want to support python 3.5+

Reply:
@Legorooj Since uvicorn is working only with python 3.6+ I have added [python_version > '3.5'](https://github.com/pyinstaller/pyinstaller/pull/4664/commits/3017d80a6419d83609c3c9f8522afa41d8a070e1) to requirements-libraries.txt and used [@importskip decorator](https://github.com/pyinstaller/pyinstaller/pull/4664/commits/af12f647f5792150fdcc457f56d7b1d3600eb0c7) in test_uvicorn.py.
Since this test is skipped on python 3.5, and test case is example copied from uvicorn docs, do you really want me to change f-strings to `.format()`?

Reply:
@genadijrazdorov never mind; if you've got the proper decorators it's fine - unless hartmut says otherwise?